### PR TITLE
Implement GameWindowTransition stuff and the remaining bits of

### PR DESCRIPTION
**Pull request by jonwil**
WindowLayout

### CODE REVIEW

File: src/game/client/gui/gamewindowtransitions.cpp [hunk 1]
```diff
@@ -13,11 +13,229 @@
...
+    }
+
+    m_transition = nullptr;
+}
+
+bool TransitionWindow::Init()
+{
+    m_winNameKey = g_theNameKeyGenerator->Name_To_Key(m_winName.Str());
+    m_window = g_theWindowManager->Win_Get_Window_From_Id(nullptr, m_winNameKey);
+    m_unk1 = m_frameDelay;
```
Comment (xezon) [inline src/game/client/gui/gamewindowtransitions.cpp:57]: `m_startFrame`

File: src/game/client/gui/gamewindowtransitions.cpp [hunk 2]
```diff
@@ -13,11 +13,229 @@
...
+{
+    if (window != nullptr) {
+        m_windows.push_back(window);
+    }
+}
+
+// clang-format off
+FieldParse GameWindowTransitionsHandler::s_gameWindowTransitionsFieldParseTable[] = { 
+    { "Window", &GameWindowTransitionsHandler::Parse_Window,  nullptr,  0 },
+    { "FireOnce", &INI::Parse_Bool,  nullptr,  offsetof(TransitionGroup, m_fireOnce) },
```
Comment (xezon) [inline src/game/client/gui/gamewindowtransitions.cpp:211]: There are double spaces in these lines

File: src/game/client/gui/windowlayout.cpp [hunk 3]
```diff
@@ -104,21 +105,57 @@ GameWindow *WindowLayout::Find_Window(GameWindow *window)
...
+            }
+
+            m_filenameString = filename;
+            Set_Init(layout.m_initFunc);
+            Set_Update(layout.m_updateFunc);
+            Set_Shutdown(layout.m_shutdownFunc);
+            return true;
+        } else {
+            captainslog_dbgassert(false, "WindowLayout::load - Failed to load layout");
+            captainslog_debug("WindowLayout::load - Unable to load layout file '%s'", filename.Str());
```
Comment (xezon) [inline src/game/client/gui/windowlayout.cpp:157]: ::Load

File: src/game/client/gui/gamewindowtransitions.cpp [hunk 4]
```diff
@@ -13,11 +13,229 @@
...
+}
+
+TransitionGroup::TransitionGroup() : m_fireOnce(false), m_currentFrame(0) {}
+
+TransitionGroup::~TransitionGroup()
+{
+    for (auto it = m_windows.begin(); it != m_windows.end(); it++) {
+        TransitionWindow *window = *it;
+
+        if (window != nullptr) {
```
Comment (feliwir) [inline src/game/client/gui/gamewindowtransitions.cpp:124]: safe to delete nullptr

File: src/game/client/gui/gamewindowtransitions.cpp [hunk 5]
```diff
@@ -13,11 +13,229 @@
...
+{
+    m_group1 = nullptr;
+    m_group2 = nullptr;
+    m_group3 = nullptr;
+    m_group4 = nullptr;
+
+    for (auto it = m_groupList.begin(); it != m_groupList.end(); it++) {
+        TransitionGroup *group = *it;
+
+        if (group != nullptr) {
```
Comment (feliwir) [inline src/game/client/gui/gamewindowtransitions.cpp:231]: safe to delete nullptr

File: src/game/client/gui/gamewindowtransitionsstyles.cpp [hunk 6]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @file
+ *
+ * @author tomsons26
```
Comment (feliwir) [inline src/game/client/gui/gamewindowtransitionsstyles.cpp:4]: Did you forget to update the author?

**Review by feliwir — APPROVED** [review]

**Pull request merged**